src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -469,9 +473,12 @@ def get_access_token_for_resource(self, username, tenant, resource):
 
     @staticmethod
     def _try_parse_msi_account_name(account):
-        subscription_name, user = account[_SUBSCRIPTION_NAME], account[_USER_ENTITY].get(_USER_NAME)
+        msi_info, user = account[_USER_ENTITY].get(_ASSIGNED_IDENTITY_INFO), account[_USER_ENTITY].get(_USER_NAME)
+
         if user in [_SYSTEM_ASSIGNED_IDENTITY, _USER_ASSIGNED_IDENTITY]:
-            parts = subscription_name.split('-', 1)
+            if not msi_info:
+                msi_info = account[_SUBSCRIPTION_NAME]  # fall back to old persisting way
```

Comment:
good question. the code is a bit messy here to support both old and new styles. 
The `msi_info` will be empty if the account is either regular user account or service principal account, which has the `user.assignedIdentityInfo` at empty. You can run `az account show` to get reference. It will also be empty if the MSI account was established using older CLI version, which leverages "subscription name" to persist managed identity info. This turned a few users off because a subscription has its own name which they want to keep, so they asked me to fix that. So I submitted this PR :). 
And yes, the value will be `MSI` for system assigned identity and will be `MSI-<id>` for user assigned identity which is the existing convention.